core/lib/presentation/utils/theme_utils.dart

```diff
@@ -6,12 +6,13 @@ import 'package:flutter/services.dart';
 class ThemeUtils {
   ThemeUtils._();
 
-  static ThemeData get appTheme {
+  static ThemeData buildAppTheme(BuildContext context) {
```

Comment:
```suggestion
  static ThemeData buildAppTheme(ThemeData themeData) {
```

IMO we should avoid passing `BuildContext` when not necessary.